sql/catalyst/src/main/java/org/apache/spark/sql/connector/util/V2ExpressionSQLBuilder.java

```diff
@@ -36,8 +37,12 @@ public String build(Expression expr) {
       return visitFieldReference((FieldReference) expr);
     } else if (expr instanceof GeneralScalarExpression) {
       GeneralScalarExpression e = (GeneralScalarExpression) expr;
+      List<String> children;
       String name = e.name();
       switch (name) {
+        case "IN":
+          children = Arrays.stream(e.children()).map(c -> build(c)).collect(Collectors.toList());
```

Comment:
nit: it seems cleaner to make `children` a local variable here: `List<String> children = Arrays.stream...`